pkg/services/sqlstore/alert.go

```diff
@@ -94,7 +94,12 @@ func HandleAlertsQuery(query *m.GetAlertsQuery) error {
 			if i > 0 {
 				sql.WriteString(" OR ")
 			}
-			sql.WriteString("state = ? ")
+			if strings.HasPrefix(v, "not") {
+				sql.WriteString("state <> ? ")
+				v = v[4:]
```

Comment:
I think its more explicit to use strings.Replace Ex
```go
v = strings.Replace(v, "not", "", 1)
```

Reply:
but that is less correct, if we ever add a state with the char pattern "not" inside it. 

Maybe adding a comment is enough "// remove not_ prefix" , go does not have a strings.Substring, as that is done via slice syntax like above

Reply:
totally right

A better solution would be
```go
v = strings.TrimPrefix(v, "not")
```
https://golang.org/src/strings/strings.go?s=21287:21327#L806

Reply:
agree! will fix and merge